lib/libc/sys/reboot.2
.\"	BSDI reboot.2,v 2.1 1995/02/03 06:51:04 polk Exp
.\"
.\" Copyright (c) 1980, 1991, 1993
.\"	The Regents of the University of California.  All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. All advertising materials mentioning features or use of this software
.\"    must display the following acknowledgement:
.\"	This product includes software developed by the University of
.\"	California, Berkeley and its contributors.
.\" 4. Neither the name of the University nor the names of its contributors
.\"    may be used to endorse or promote products derived from this software
.\"    without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
.\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
.\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
.\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
.\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
.\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
.\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
.\" SUCH DAMAGE.
.\"
.\"     @(#)reboot.2	8.1 (Berkeley) 6/4/93
.\"
.Dd June 4, 1993
.Dt REBOOT 2
.Os BSD 4
.Sh NAME
.Nm reboot
.Nd reboot system or halt processor
.Sh SYNOPSIS
.Fd #include <unistd.h>
.Fd #include <sys/reboot.h>
.Ft int
.Fn reboot "int howto"
.Sh DESCRIPTION
.Fn Reboot
reboots the system.
Only the super-user may reboot a machine on demand.
However, a reboot is invoked
automatically in the event of unrecoverable system failures.
.Pp
.Fa Howto
is a mask of options; the system call interface allows the following
options, defined in the include file
.Aq Pa sys/reboot.h ,
to be passed
to the new kernel or the new bootstrap and init programs.
.Bl -tag -width RB_INITNAMEA
.It Dv RB_AUTOBOOT
The default, causing the system to reboot in its usual fashion.
.It Dv RB_ASKNAME
Interpreted by the bootstrap program itself, causing it to
prompt on the console as to what file should be booted.
Normally, the system is booted from the file
.Dq Em xx Ns No (0,0)bsd ,
where
.Em xx
is the default disk name,
without prompting for the file name.
.It Dv RB_DFLTROOT
Use the compiled in root device.
Normally, the system uses the device from which it was booted
as the root device if possible.
(The default behavior is dependent on the ability of the bootstrap program
to determine the drive from which it was loaded, which is not possible
on all systems.)
.It Dv RB_DUMP
Dump kernel memory before rebooting; see
.Xr savecore 8
for more information.
.It Dv RB_HALT
the processor is simply halted; no reboot takes place.
This option should be used with caution.
.It Dv RB_INITNAME
An option allowing the specification of an init program (see
.Xr init 8 )
other than
.Pa /sbin/init
to be run when the system reboots.
This switch is not currently available.
.It Dv RB_KDB
Load the symbol table and enable a built-in debugger in the system.
This option will have no useful function if the kernel is not configured
for debugging.
Several other options have different meaning if combined
with this option, although their use may not be possible
via the
.Fn reboot
call.
See
.Xr kadb 4
for more information.
.It Dv RB_NOSYNC
Normally, the disks are sync'd (see
.Xr sync 8 )
before the processor is halted or rebooted.
This option may be useful if file system changes have been made manually
or if the processor is on fire.
.It Dv RB_RDONLY
Initially mount the root file system read-only.
This is currently the default, and this option has been deprecated.
.It Dv RB_SINGLE
Normally, the reboot procedure involves an automatic disk consistency
check and then multi-user operations.
.Dv RB_SINGLE
prevents this, booting the system with a single-user shell
on the console.
.Dv RB_SINGLE
is actually interpreted by the
.Xr init 8
program in the newly booted system.
.Pp
When no options are given (i.e.,
.Dv RB_AUTOBOOT
is used), the system is
rebooted from file ``bsd'' in the root file system of unit 0
of a disk chosen in a processor specific way.
An automatic consistency check of the disks is normally performed
(see
.Xr fsck 8 ) .
.El
.Sh RETURN VALUES
If successful, this call never returns.
Otherwise, a -1 is returned and an error is returned in the global
variable
.Va errno .
.Sh ERRORS
.Bl -tag -width Er
.It Bq Er EPERM
The caller is not the super-user.
.El
.Sh SEE ALSO
.Xr kadb 4 ,
.Xr crash 8 ,
.Xr halt 8 ,
.Xr init 8 ,
.Xr reboot 8 ,
.Xr savecore 8
.Sh BUGS
The HP300 implementation supports neither
.Dv RB_DFLTROOT
nor
.Dv RB_KDB .
.Sh HISTORY
The
.Nm
function call appeared in
.Bx 4.0 .


lib/libc/sparc/sys/reboot.s
/*
 * Copyright (c) 1992, 1993
 *	The Regents of the University of California.  All rights reserved.
 *
 * This software was developed by the Computer Systems Engineering group
 * at Lawrence Berkeley Laboratory under DARPA contract BG 91-66 and
 * contributed to Berkeley.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. All advertising materials mentioning features or use of this software
 *    must display the following acknowledgement:
 *	This product includes software developed by the University of
 *	California, Berkeley and its contributors.
 * 4. Neither the name of the University nor the names of its contributors
 *    may be used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 *
 * from: /master/lib/libc/sparc/sys/reboot.s,v 2.1 1995/02/03 06:39:52 polk Exp
 */

#if defined(LIBC_SCCS) && !defined(lint)
	.asciz "@(#)reboot.s	8.1 (Berkeley) 6/4/93"
#endif /* LIBC_SCCS and not lint */

#include "SYS.h"

SYSCALL(reboot)
	unimp	0
